src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -1835,6 +1887,280 @@ public async Task<bool> HasSecretsManagerStandalone(Organization organization)
+                        parameters.TaxInformation.TaxId,
+                        parameters.TaxInformation.Country);
+                    throw new BadRequestException("billingPreviewInvalidTaxIdError");
+                default:
+                    _logger.LogError(e, "Unexpected error previewing invoice with tax ID '{TaxId}' in country '{Country}'.",
+                        parameters.TaxInformation.TaxId,
+                        parameters.TaxInformation.Country);
+                    throw new BadRequestException("billingPreviewInvoiceError");
+            }
+        }
+    }
+
+    public async Task<PreviewInvoiceResponseModel> PreviewInvoiceAsync(
+        PreviewOrganizationInvoiceRequestBody parameters,
+        string gatewayCustomerId,
+        string gatewaySubscriptionId)
+    {
+        var plan = Utilities.StaticStore.GetPlan(parameters.PasswordManager.Plan);
+
+        var options = new InvoiceCreatePreviewOptions
+        {
+            AutomaticTax = new InvoiceAutomaticTaxOptions
+            {
+                Enabled = true,
+            },
+            Currency = "usd",
+            Discounts = new List<InvoiceDiscountOptions>(),
+            SubscriptionDetails = new InvoiceSubscriptionDetailsOptions
+            {
+                Items =
+                [
+                    new()
+                    {
+                        Quantity = parameters.PasswordManager.AdditionalStorage,
+                        Plan = plan.PasswordManager.StripeStoragePlanId
+                    }
+                ]
+            },
+            CustomerDetails = new InvoiceCustomerDetailsOptions
+            {
+                Address = new AddressOptions
+                {
+                    PostalCode = parameters.TaxInformation.PostalCode,
+                    Country = parameters.TaxInformation.Country,
+                }
+            },
+        };
+
+        switch (plan.ProductTier)
+        {
+            case ProductTierType.Families:
+                options.SubscriptionDetails.Items.Add(
+                    new()
+                    {
+                        Quantity = 1,
+                        Plan = plan.PasswordManager.StripePlanId
+                    }
+                );
+                break;
+            case ProductTierType.Teams:
+            case ProductTierType.TeamsStarter:
+            case ProductTierType.Enterprise:
+                options.SubscriptionDetails.Items.Add(
+                    new()
+                    {
+                        Quantity = parameters.PasswordManager.Seats,
+                        Plan = plan.PasswordManager.StripeSeatPlanId
+                    }
+                );
+                break;
+        }
+
+        if (plan.SupportsSecretsManager)
+        {
+            options.SubscriptionDetails.Items.AddRange([
+                new()
+                {
+                    Quantity = parameters.SecretsManager?.Seats ?? 0,
```

Comment:
🎨 This section doesn't use the above switch-case, but you have a similar option on the `SecretsManagerPlanFeatures` to use `HasAdditionalSeatsOption` and `HasAdditionalServiceAccountOption`, if you wish